FlagsmithClient/Classes/Flagsmith.swift

```diff
@@ -12,269 +12,291 @@ import FoundationNetworking
-        completion(.success(hasFlag))
-      case .failure(let error):
-        if self.defaultFlags.contains(where: {$0.feature.name == id && $0.enabled}) {
-          completion(.success(true))
+
+    /// Configuration class for the cache settings
+    private var _cacheConfig: CacheConfig = CacheConfig()
+    public var cacheConfig: CacheConfig  {
+        get {
+            apiManager.propertiesSerialAccessQueue.sync { _cacheConfig }
         }
-        else {
-          completion(.failure(error))
+        set {
+            apiManager.propertiesSerialAccessQueue.sync {
+                _cacheConfig = newValue
+            }
         }
-      }
     }
-  }
-  
-  /// Get remote config value optionally for a specific identity
-  ///
-  /// - Parameters:
-  ///   - id: ID of the feature
-  ///   - identity: ID of the user (optional)
-  ///   - completion: Closure with Result which String in case of success or Error in case of failure
-  @available(*, deprecated, renamed: "getValueForFeature(withID:forIdentity:completion:)")
-  public func getFeatureValue(withID id: String,
-                              forIdentity identity: String? = nil,
-                              completion: @escaping (Result<String?, Error>) -> Void) {
-    analytics.trackEvent(flagName: id)
-    getFeatureFlags(forIdentity: identity) { (result) in
-      switch result {
-      case .success(let flags):
-        let flag = flags.first(where: {$0.feature.name == id})
-        completion(.success(flag?.value.stringValue))
-      case .failure(let error):
-        if let flag = self.getFlagUsingDefaults(withID: id, forIdentity: identity) {
-          completion(.success(flag.value.stringValue))
+
+    private init() {
+        apiManager = APIManager()
+        analytics = FlagsmithAnalytics(apiManager: apiManager)
+    }
+
+    /// Get all feature flags (flags and remote config) optionally for a specific identity
+    ///
+    /// - Parameters:
+    ///   - identity: ID of the user (optional)
+    ///   - completion: Closure with Result which contains array of Flag objects in case of success or Error in case of failure
+    public func getFeatureFlags(forIdentity identity: String? = nil,
+                                completion: @Sendable @escaping (Result<[Flag], any Error>) -> Void) {
+
+        if let identity = identity {
+            getIdentity(identity) { (result) in
+                switch result {
+                case .success(let thisIdentity):
+                    completion(.success(thisIdentity.flags))
+                case .failure(let error):
+                    if self.defaultFlags.isEmpty {
+                        completion(.failure(error))
+                    }
+                    else {
+                        completion(.success(self.defaultFlags))
+                    }
+                }
+            }
+        } else {
+            apiManager.request(.getFlags) { (result: Result<[Flag], Error>) in
```

Comment:
i.e. 

```suggestion
            apiManager.request(.getFlags) { (result: Result<[Flag], any Error>) in
```

Reply:
In this instance we're just telling the compiler the type of the Result class we expect, rather than defining the API. request() is generic, so the compiler needs a heads-up:

```swift
    func request<T: Decodable>(_ router: Router, using decoder: JSONDecoder = JSONDecoder(),
                               completion: @Sendable @escaping (Result<T, any Error>) -> Void)
```